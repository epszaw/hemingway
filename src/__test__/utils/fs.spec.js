const path = require('path')
const { readDir, readFile } = require('src/utils/fs')

describe('utils â€“ fs', () => {
  const FIXTURES_PATH = path.resolve(__dirname, '../__fixtures__')

  describe('readDir', () => {
    it('should returns folder content', async () => {
      expect.assertions(1)

      const res = await readDir(FIXTURES_PATH)

      expect(res).toEqual(['stories', 'trees'])
    })
  })

  describe('readFile', () => {
    it('should returns file content in given encoding', async () => {
      expect.assertions(1)

      const res = await readFile(
        path.join(FIXTURES_PATH, './stories/common_story.md'),
        'utf8'
      )

      expect(res).toMatchSnapshot()
    })
  })
})
