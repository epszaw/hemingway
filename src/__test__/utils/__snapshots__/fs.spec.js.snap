// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils – fs readDir should returns folder content 1`] = `
Array [
  "stories",
  "trees",
]
`;

exports[`utils – fs readFile should returns file content in given encoding 1`] = `
"OPEN \\"https://google.com\\"
FIND \\"#lst-ib\\" AS \\"search-input\\"
TAKE \\"search-input\\" TYPE \\"Hello{Enter}\\"
WAIT \\"#rhscol\\"
FIND \\".bkWMgd\\" HAVE \\"length\\" NOT EQUAL \\"0\\"
"
`;
